src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala

```diff
@@ -91,27 +95,51 @@ class BlockFetcher(
 
   private def handleHeadersMessages(state: BlockFetcherState): Receive = {
     case Response(peer, BlockHeaders(headers)) if state.isFetchingHeaders =>
-      log.debug("Fetched {} headers starting from block {}", headers.size, headers.headOption.map(_.number))
-
-      val newState = state.validatedHeaders(headers) match {
-        case Left(err) =>
-          peersClient ! BlacklistPeer(peer.id, err)
-          state.fetchingHeaders(false)
-        case Right(validHeaders) =>
-          state.appendHeaders(validHeaders)
-      }
+      val newState =
+        if (state.fetchingHeadersState == AwaitingHeadersToBeIgnored) {
```

Comment:
Maybe I am missing somehing, but what will happen in case of request time out ? 
I see such path:

1. request headers
2. fetcher receive invalidate
3. request times out, but we are still in `AwaitingHeadersToBeIgnored` state
4. fetcher create request for new headers, from new invalidated blocks
5. fetcher receivers response witth new block headers, but it is ignored as fetcher is still in  `AwaitingHeadersToBeIgnored`

Is it correct, or there is some detail i am missing ?

Reply:
You are right! I thought I had analyzed that but I missed it

I'll update this PR with something even closer to what we do when receiving the response